src/shims/env.rs

```diff
@@ -321,20 +322,27 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             "`getcwd` is only available for the UNIX target family"
         );
 
-        this.check_no_isolation("`getcwd`")?;
-
-        let buf = this.read_scalar(&buf_op)?.check_init()?;
-        let size = this.read_scalar(&size_op)?.to_machine_usize(&*this.tcx)?;
-        // If we cannot get the current directory, we return null
-        match env::current_dir() {
-            Ok(cwd) => {
-                if this.write_path_to_c_str(&cwd, buf, size)?.0 {
-                    return Ok(buf);
+        match this.machine.isolated_op {
+            IsolatedOp::Allow => {
+                let buf = this.read_scalar(&buf_op)?.check_init()?;
+                let size = this.read_scalar(&size_op)?.to_machine_usize(&*this.tcx)?;
+                // If we cannot get the current directory, we return null
+                match env::current_dir() {
```

Comment:
I'd prefer if we could avoid the rightwards drift here.
Can you handle the "reject" case early and return, so that the rest of the function can proceed as before?